File: Python/SivitasAkademik.py
from Human import Human

class SivitasAkademik(Human):
    __asal_universitas = ""
    __email_edu = ""

    def __init__(self, nik="", nama="", jk='',  asal_univ="", email_edu = ""):
        super(SivitasAkademik, self).__init__(nik, nama, jk)
        self.__asal_universitas = asal_univ
        self.__email_edu = email_edu

    def setAsalUniv(self, asal_univ):
        self.__asal_universitas = asal_univ

    def setEmailUniv(self, email_univ):
        self.__email_edu = email_univ

    def getAsalUniv(self):
        return self.__asal_universitas
    
    def getEmailUniv(self):
        return self.__email_edu
    

File: CPP/SivitasAkademik.cpp
#include "SivitasAkademik.h"

SivitasAkademik::SivitasAkademik()
{
    this->asal_universitas = "";
    this->email_edu = "";
}

SivitasAkademik::SivitasAkademik(string asal_universitas, string email_edu)
{
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
}

SivitasAkademik::SivitasAkademik(string nik, string nama, char jk, string asal_universitas, string email_edu) : Human(nik, nama, jk)
{
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
}

void SivitasAkademik::setAsalUniv(string asal_universitas)
{
    this->asal_universitas = asal_universitas;
}

void SivitasAkademik::setEmailUniv(string email_edu)
{
    this->email_edu = email_edu;
}

string SivitasAkademik::getAsalUniv()
{
    return this->asal_universitas;
}

string SivitasAkademik::getEmailUniv()
{
    return this->email_edu;
}

SivitasAkademik::~SivitasAkademik() {}